lib/core.ps1

```diff
@@ -103,7 +103,16 @@ function url_remote_filename($url) {
     split-path (new-object uri $url).absolutePath -leaf
 }
 
-function ensure($dir) { if(!(test-path $dir)) { mkdir $dir > $null }; resolve-path $dir }
+function ensure($dir) {
```

Comment:
How about a unix_compat.ps1 file that conditionally overrides core functions? And potentially other functions. It could be imported only when needed in order to keep core.ps1 clean, and it would just need one 'if' test for unix.